lib/mapper/product_mapper.rb

```diff
@@ -158,6 +160,24 @@ def self.get_variant_by_sku(sku, product)
                 product['model'].select {|m| m['sku'] == sku}.first
             end
 
+            def self.merge_additional_properties(clone, field_map)
+                if clone['model'].select {|a| a['additionalProperty']}
+                    additionalProperty = clone['model'].select {|a| a['additionalProperty']}
+                    additionalProperty.each do |prop|
+                        if clone['model'].select {|a| a['isDefault']}
+                            prop.select {|a| a['additionalProperty']}['additionalProperty'].each do |a|
+                                if field_map && field_map['additionalProperty']
+                                    field_map['additionalProperty'].each do |field, key|
+                                        if a['propertyID'] == field[key]
+                                            clone['additionalProperty'].push(a)
+                                        end
+                                    end
+                                end
+                            end
+                        end
+                    end
+                end
+            end
```

Comment:
@Jhalter5Stones From an efficiency standpoint, I would probably _start_ with the values in `fieldMap`:

```
defaultVariant = clone['model'].select { |v| v['isDefault'] }
if defaultVariant && defaultVariant['additionalProperty']
    field_map['additionalProperty'].each do |field, key|
        prop = defaultVariant['additionalProperty'].select { |prop| prop['propertyID'] == field[key] }
        clone['additionalProperty'].push(prop) unless prop.nil?
    unless  field_map.nil? || field_map['additionalProperty'].nil?
end
```